### DISCUSSION TITLE
Higher Reward for early Submission of Flips

### DISCUSSION THREAD

**Discussion by ubiubi18** [idena-network/idena-docs#86] [IIPs]
# Idea: 

 Only flips that are submitted within first 8 hours of each epoch should be rewarded. This will result in higher rewards for unique people and more AI-resistant flips.

 (Update may 2024: additional make the creation of one flip mandative within first hour of new epoch, if you want to keep your right to run a mining node. Identities status is not affected)

# Motivation:

As long as there are no time-constraints submitting flips, big players have legit incentive to centralise their flip-production to be able to extract value more efficient and try to get it done as fast as possible. This leads also to bad practice. Story-template flips ( - for example starting every valid story with someone waking up in the first picture and adding some action in the last three pictures for the two keywords -) can be produced in two minutes and reach more often consensus than genuine flips that take much longer to produce. Unfortunately those repeating story-patterns are less AI-resistant. There are likely other undesirable strategies of big players that enfavor possible sybil attacks and are not covered by the report system. Introducing time-contraints to rewardable submission of flips will make it less profitable to centraly produce flips and favor solo-miners and family-pools over dishonest players.

# Technical specification:

If a flip has been submitted within the first 1440 blocks (about 8 hours) of  the epoch and if it does get qualified and is not reported by a majority, a basic flip reward  is payed. The basic reward is calculated based on how early in the epoch the flip was submitted. The later the flip is submitted, the lower the basic flip reward. The reduction factor k is calculated for each submitted flip as k = 1−t⁴ · 0.5, where t ∈ [0..1] is the amount of time that has passed from the start of the epoch to the moment of the submission until 1440 new blocks are mined. Flips can also be submitted more than 1440 blocks into the new epoch, but the submitter will receive zero rewards for those flips.

# Rationale:

Even a human farm that is able to create one flip as fast as every two minutes can only produce flips for up to 10 identities each hour, but not for hundreds at once. That player would need to involve more real people instead, if the task was to create flips for a farm with more than a few dozen identities within 8 hours.

Unique people validating (semi-)unique identities are so far mainly required in two session tasks: reporting and solving flips. Those tasks cant be centralised or automatised and secure idenas semi-unique identity. No matter that they are only responsible for 25% of possible rewards in idenanetwork, while the other 75% of total rewards are extracted in tasks that are prone to centralisation. 

According to the block explorer human farms usualy produce flips in a row over the span of the whole epoch and the fastest of them use strategies like reusing flips or using story-templates to achieve that. Also the badest actors are most competitive to survive low marketcap, since their production costs each flip are smaller. 

Solo-miners and honest pools become faster unprofitable in flip production than dishonest players, so „free market“ logic unfortunately enfavors bad habits. 

The majority of unique people involved in idena flip-sessions  does not even have the chance to earn rewards and secure idena by producing flips! This can be deemed as a security risk alone and makes it at the same time cryptonomically less profitable to be a solo-miner, because big pools that centralise flip-production are faster in producing and submitting flips and thereby lower the median market value of that task. 

If we we start to reward only flips that are submitted on the first day of each epoch, big players will need to decide to involve more people to get it done in time or to loose flip rewards.
Involving more people will lift production costs for human farms substancially in favor of unique people, spread rewards to more people and make the production of AI-resistant semi-unique flips more likely. 

Story-templates can still be used in this setting, but the risk/profit ratio will be more undesirable for big players than it is now. Also it is less expensive to introduce more creative stories, if a farm has to involve and pay more people anyway. More farm workers beeing involved into flip production should also make it more realistic, that workers decide to turn into solo-miners.

There are several alternative approaches we could try to deal with bad habbits of centralised flip producers:

- Appeal for good will, teach them about implications of their behaviour

I think idena has an impressing history of teaching people about best practice. This was absolute worth it! Some big pools do a brilliant job in producing good flips that are sybil-resistant no matter that they are produced centrally. But at this point it seems unlikely that other big players with bad habbits are unaware of the facts or do even care. So appealing for more good will might be ineffective.

- Lower or even cut all flip rewards to zero

This is neither compatible with the upcoming incentive realignment of grading flips, nor is it desirable in context of staking, because it will make it more expensive for late adopters to collect the required idna for their stake just with work.

- Hope for flip grading to solve the issue

Flip grading is great but wont cover every aspect of centralised flip production. There will allways be undesirable patterns possible in flip production that are not instantly visible for the flip reporting committee. 

It may seem more user-friendly to give honest people as much time as possible to come up with good flips instead of producing them in a hurry, it may also seem convenient to outsource the process of flipproduction to people or centralised players who know how to do it better and faster – but lowering and optimising the production costs of big players ultimately hurts a small player more than the risk of loosing a few flip rewards due to the lack of time. If people have about eight rewardable submission hours, that should in most cases be enough time to arrange a submission. If people dont find the time, they will still have their staking rewards, which will outweight the potential flip rewards losses in most cases.

# Backwards Compatibility 

should be comparable to the change that introduced a reduction factor over time to the invitation rewards

# Reference Implementation 

the proposal leans to the reduction factor of the invitation reward fund, which is proven to work just fine: https://docs.idena.io/docs/wp/economics#invitation-reward-fund

# Security Considerations:

This IIP is at risk to be opposed by a majority of eligable identities onchain, because pools that are bigger than a family will have problems to cover the time limitation with centralised production routines, while pools smaller than that as well as unique solo miners might not all see the neccessity and oppose it as well for fearing to loose rewards too. 

If implemented right now at small marketcap, it could also result in an even faster decline of network size in short term, because many pools would need to rearange their strategy or even question their whole profitability of their operations under new conditions. Some of them could shut down operations and sell, while at the same time some solo-miners would feel frustrated about not finding the time for flip production and loosing even those rewards.

Also it is undesirable to loose too many farms, they are managed by competent people who understand idena and that do a hell of a job in introducing idena to new humans. It should not be the goal to shut all pools down, just because a few of them have bad habbits.

But i think the medium term benefits should outweight that risk. 

The task of solving flips secures itself with a time-limit against sybil-attacks, the task of producing flips should do the same. Centralisation of flip-production poses a constant security risk, because it is prone to AI-recognition and other abuse. For better sybil-resistance flips need to be produced as semi-unique as possible. As more people are involved in producing flips, as more unique creativity can be involved without additional costs, which will probably be recognisable much harder for possible AI-assistance in solving flips.

**Comment by Zen-44** [comment]
I am not sure if the exact invitation decreasing rewards formula is fit for this purpose as well. Invitation rewards are not meant to drop to 0 at any point so the function decreases to 0.5. Using it in this scenario will lead to flip rewards decreasing to 0.5, and then a sharp decrease to 0 in terms of rewards in the span of one block. A formula decreasing to 0 should be used instead.

This can be achieved by simply removing `*0.5` from the formula, resulting in `k=1-t^4`.

Another detail that should be considered is that the current formula used for decreasing invitation rewards is used for a bigger interval (currently 21 days). The formula has a steeper drop that starts later in the interval. It may be worth considering raising `t` to the power of `3` (`k=1-t^3`). It will have a more linear decline and will start to get pronounced earlier in the interval. It might be more fit for an interval of just 8 hours. Some other formula could be used to achieve something similar.

**Reply by ubiubi18** [reply]
thank you, zen. Yes, this formula could look different. Though i think it is good to have the first few hours with well above 80% rewards, so that family pools or people who need to sleep or to work dont are discriminated too much and can still keep on doing it kind of the old way? That is an argument against t to the power of 3 ?

**Comment by ubiubi18** [comment]
I would like to add that time-constraint submission will make it harder to profitably use AI-assistance in flip production in the future. If a player eventually finds a way to get completely rid of the requirement of human work in creating flips, it will still take him/her time to compute keywords with a storyteller-AI and later do automatically websearches for good pictures or use something like DALL E to create needed pictures.  If we have a time-limit of a few hours, this process would boil down to either not beeing sufficient to supply a pool with enough AI-flips in time, or beeing a pretty expensive task where the costs probably outweight the benefits when you have to rent multiple instances of AI-service at once to speed it up like in a proof of work. Luckily we are not there yet anyway.

**Comment by Zen-44** [comment]
Before all, it should be considered that flip rewards represent a large chunk of rewards, especially for new members. It will make it harder to get into staking without external investments for newbies if they are constrained to only get rewards if they submit flips in some amount of time. 

With IIP-5 newbies will have to put in a lot of work to get started and get into staking to be able to mine affordably. The previous argument of newbies just being able to validate and create good flips will no longer be valid. For the current people involved in the network, it probably won't be a big loss, with decent staking you can earn x times more than the rewards from flips with staking. Though the changes might be asking too much from newbies considering 3-4 qualified flips may represent most of their rewards at young ages.

This is a negative aspect from the perspective of an average new member. With enough dedication, this won't affect newbies either too much, but how many will be dedicated enough? (Of course, considering 0 investments upfront)

**Reply by ubiubi18** [reply]
i understand that more as an argument against IIP-5 in general, but not as an argument especially against more decentralised flip production. Also i would expect flip rewards to grow for those newbies who make it in time, since there will be allways people who dont make it in time but produce flips anyway - especially people with higher stakes or staking pools. So newbies will have less competitors for the flip reward pool, newbies can grow their stake even faster with just a bit more timing required in this setting. And every unique human has the same chance to achieve that, no matter how big the stake is.

**Reply by ubiubi18** [reply]
By the way, it is a good hint that flip rewards are less important if you have a higher stake. So most early adopters with bigger stakes dont really need to fear it coming too late, but can even see it as a good way to support all newbies intentionally, if they want. A small luxury of people with high stake doing flips only after a few days just to keep the identity alive, but not in a hurry for the relativly marginal flip rewards.

**Reply by Zen-44** [reply]
Not an argument against any of these. I am trying to look over what the ramifications of the methods we choose will be. Poking a stick into every aspect to not leave things undiscussed. I support decentralized flip production. 👍 
It's also true that on the other side, the people who can and do put in the time will get a lot more coins since the flip reward fund won't cover 38k flips anymore, just the ones submitted in said timeframe. That's a good point.

**Reply by ubiubi18** [reply]
@midenaio A technical question: would it theoretically be possible to change the session mechanics in a way that keywords for next epoch are allready distributed between short session and long session of current epoch? 

So people could use the waiting time while the network is reaching consensus about the validated identities to prepare new flips for the next epoch on their own risk? Though not knowing yet if they are eligable to submit them or not? Or would there be a problem in generating the keywords that early? Or with people trying to submit flips too early, while the network load should still focus mainly on the session?

I think if the keywords would show up earlier, that would make it more likely and cheaper for pools to involve more unique people into flip production right away instead of doing it centralised.

**Reply by midenaio** [reply]
@ubiubi18 it's technically possible to change the block in which the seed for the flips keywords is generated. However submitting future flips for the next epoch during the long session would be problematic.

**Comment by ubiubi18** [comment]
I would like to thank Dalik to adding something to that discussion

![dalik](https://user-images.githubusercontent.com/39676643/182192374-e77e4ac6-bdc5-49db-a978-35ab4db918f8.JPG)

 (Disclaimer: I think Daliks 2 known pools do reasonable good flips, i see no big problem in those pools. Also he could adopt this IIP much easier, since most of his identities have higher stake allready due to their age. So this IIP-idea aims at other players. )

**Comment by 0JustinMiles0** [comment]
At first, I was skeptical of this proposal because I was concerned that the time window to create flips would be short and right after the validation of the identities by the network which would require participants to dedicate about an hour of their time every 21 days rather than a few minutes. Also, I was concerned that a short time window to create flips could impair the quality of the flips produced by non-farms participants. However, I think this mechanism could be an effective way to combat mass-production of flips by farms. At the minimum it would substantially increase their flips production costs. Below are the elements I'd like to see preserved by this proposal:

- Possibility to submit flips any time during the epoch (however later flips submissions won't pay or not as much).
- A time window after the ceremony that pays 100% of the flip rewards for flip submissions. I would like to see this time window longer than 8 hours. I think that 24 hours should be the minimum. 
- A progressive decay of the flip rewards paid past this time window. We should not go from 100% before 24 hours to 0% after 24 hours. Honest participants may not be available to create flips after the ceremony for several hours/days. A 0% payment after 24 hours seems too harsh. 

I'm not sure what would be the ideal decay function but I would have imagined something like this:
<img align="middle" src="https://user-images.githubusercontent.com/109244770/188262257-e08bceef-b160-487b-bd08-b44875012a10.png"/><br>

A faster decay that would reach 50% of the flips rewards before 50% of the epoch length is reached is probably fine.

This proposal combined to a flips grading mechanism could be an effective way to severely undermine the economic model of flips farms.

**Reply by ubiubi18** [reply]
@midenaio I wont submit this IIP-proposal, because i think the path of quadratic staking is more promissing at this time. I just wonder if the idea of time-constrains to flip rewards could be combined with possible future quadratic flip rewards? Maybe that would be more convenient and give newbies better chances to collect more rewards early than without time constrains? It would be great if you could consider it, once the team starts to discuss making flip rewards quadratic too =p Thanks.

**Reply by ubiubi18** [reply]
two years later the issue of oligopoly and template flips still haunts idenanetwork. 

Yesterday we had a discussion at discord if it would help to make the production of one flip mandative during long session !

I had ChatGPT helping me to summarize the argument in favor of that, also relating to your research paper:

->:requiring each identity to create a FLIP during validation sessions can mitigate the risks of oligopoly, it's also important to consider the economic impact this requirement would have on the cost structures of identity farming operations versus genuine participants:

1. **Increased Operational Costs for Puppeteers**: The requirement for each identity to create a FLIP significantly raises the operational costs for puppeteers or identity farmers. Since each controlled identity must not only solve but also create a FLIP, the time and cognitive effort required per session doubles. Previously, a puppeteer might manage multiple identities merely by solving puzzles, which could be done relatively quickly. Now, they must also invest time in creating these puzzles, which requires a distinct set of skills and significantly more time—potentially 30 minutes per session, per identity. This makes managing large numbers of puppet identities much more labor-intensive and costly.

2. **Temporal Extension of Engagement**: The need to stay engaged with the validation process for a longer period—both to solve and then later create a FLIP—means that the puppeteer's resources are stretched thinner over time. For a genuine miner, who typically remains active in the network throughout the validation session to monitor outcomes and participate fully, this change represents little to no additional cost or inconvenience. They are already present and engaged, making the addition of FLIP creation a nominal extension of their ongoing participation.

3. **Economies of Scale Disrupted**: In the context of oligopoly, where a few entities control a large number of identities, the economies of scale that make such control profitable get disrupted. If each identity must create a FLIP, the scale of effort and coordination required grows exponentially with the number of identities controlled. This disrupts the economic advantage of scale, as each additional identity adds linearly to costs without a corresponding efficient increase in value extraction.

4. **Quality Control and Consistency Challenges**: Ensuring the quality and variety of FLIPs across multiple puppet identities becomes more challenging under this rule. Each FLIP needs to be unique and meet certain quality standards to be accepted as part of the validation process. Managing this across numerous identities increases the complexity and potential for error within a puppeteer's operations, further adding to their costs and reducing the likelihood of maintaining a covert and efficient operation.

5. **Deterring Low-Effort Participation**: By making FLIP creation mandatory, the system may deter low-effort, speculative participants who are not genuinely invested in the network's health but are rather motivated by short-term gains from simple tasks. This shifts the network dynamics towards more dedicated participants who contribute positively to the network's robustness and security.

In summary, while making FLIP creation mandatory for each identity does not eliminate the possibility of oligopoly, it introduces significant barriers that make such operations less economically viable and more risky. This helps level the playing field for genuine participants and ensures a healthier, more secure network.

<-

**Reply by ubiubi18** [reply]
I guess the idea of hlolve at discord to introduce a flipsession without mandative flipcreation but also without miningrights is compatiple with the idea to improve ogipoly-resistance at the same time for those who want to stay eligable to run a mining node

**Reply by ubiubi18** [reply]
It is also important to see it from the perspective "cost of 51% attack".  Make the privileg running a mining node more semi-unique and less oligopoly-vulnerable
